### PR TITLE
add recipe and build scripts

### PR DESCRIPTION
**Pull request by M-Waszkiewicz-Anaconda**
initial commit wasmtime-11.0.1
- dependency for wasmtime-py
- [upstream](https://github.com/bytecodealliance/wasmtime/tree/v11.0.1)

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    sha256: 65748477f6d834c2d41de6bbe773254c395e4ddeae3461f84997c58dbf6276be
+    folder: crates/wasi-nn/spec
+
+  - url: https://github.com/WebAssembly/wasi-http/archive/1c95bc21dbd193b046e4232d063a82c8b5ba7994.tar.gz
+    sha256: 8d0a31a4265f610627a86aa7766d4f511b23a19f9c06d6c9febbf29f02cb4f64
+    folder: crates/wasi-http/wasi-http
+
+build:
+  number: 0
+  skip: True # [linux and (s390x or ppc64le)]
```
Comment (akabanovs) [inline recipe/meta.yaml:30]: Perhaps add a comment why these are skipped? And those below.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,71 @@
+{% set name = "wasmtime" %}
+{% set version = "11.0.1" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
```
Comment (akabanovs) [inline recipe/meta.yaml:8]: So these are just the required components that you put into specific folders?

Comment (M-Waszkiewicz-Anaconda) [reply recipe/meta.yaml:8]: they are git submodules which are part of the project but not included in tarball

File: recipe/meta.yaml (hunk 1 above)
Comment (M-Waszkiewicz-Anaconda) [reply recipe/meta.yaml:30]: we can try to build it, but those are not supported by wasmtime-py and snowflake does not need those
see my comment on wasmtime-py
https://github.com/AnacondaRecipes/wasmtime-py-feedstock/pull/1#discussion_r1298218854

**Review by akabanovs — APPROVED** [review]

File: recipe/bld.bat [hunk 3]
```diff
@@ -0,0 +1,6 @@
+cargo build --release
+cargo build --release --manifest-path crates\c-api\Cargo.toml
+
+copy %SRC_DIR%\target\release\wasmtime.exe %LIBRARY_BIN%
+copy %SRC_DIR%\target\release\wasmtime.dll  %LIBRARY_BIN%
```
Comment (JeanChristopheMorinPerso) [inline recipe/bld.bat:5]: Does it have a companion `.lib` file?

**Review by ViridianMelody — APPROVED** [review]
LGTM!

File: recipe/build.sh [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+cargo build --release
+cargo build --release --manifest-path crates/c-api/Cargo.toml
+
+mkdir -p ${PREFIX}/bin
+mkdir -p ${PREFIX}/lib
+
+cp ${SRC_DIR}/target/release/wasmtime ${PREFIX}/bin/
+
+if [ `uname` == Darwin ]; then  
+    cp ${SRC_DIR}/target/release/libwasmtime.dylib ${PREFIX}/lib/
```
Comment (JeanChristopheMorinPerso) [inline recipe/build.sh]: We could potentially use `$SHLIB_EXT` to get the extension.

File: recipe/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    - /lib64/ld-linux-x86-64.so.2
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('rust') }} 
+
+  run:
+    - libgcc-ng                  # [linux]
+    - python
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml]: Do you really need python as a runtime dependency?

File: recipe/bld.bat (hunk 3 above)
Comment (M-Waszkiewicz-Anaconda) [reply recipe/bld.bat:5]: @JeanChristopheMorinPerso it has following files:

17.08.2023  16:48            51 315 wasmtime.d
17.08.2023  16:48        10 335 232 wasmtime.dll
17.08.2023  16:47            61 373 wasmtime.dll.exp
17.08.2023  16:47           101 722 wasmtime.dll.lib
17.08.2023  16:44        21 105 152 wasmtime.exe
17.08.2023  16:47        74 406 860 wasmtime.lib
17.08.2023  16:48         5 525 504 wasmtime.pdb

so the wasmtime.lib is static and wasmtime.dll.lib is import I believe, shall we take import only into %LIBRARY_LIB%  ?

File: recipe/build.sh (hunk 4 above)
Comment (M-Waszkiewicz-Anaconda) [reply recipe/build.sh]: yes

File: recipe/meta.yaml (hunk 5 above)
Comment (M-Waszkiewicz-Anaconda) [reply recipe/meta.yaml]: I think not, will try to remove it

File: recipe/bld.bat (hunk 3 above)
Comment (JeanChristopheMorinPerso) [reply recipe/bld.bat:5]: I frankly don't know. I'd look at how wasmtime-py links wasmtime. If everything links correctly, then we could just add a comment in the build script to say that if someone needs these files, then we could just push a PR to rebuild the package.

Comment (M-Waszkiewicz-Anaconda) [reply recipe/bld.bat:5]: wasmtime-py just uses dll only, see 
https://github.com/AnacondaRecipes/wasmtime-py-feedstock/pull/1/files#diff-6575ceeaf967f9e78afe3221eb5d5034d1edc55ffd549e28e7a1a62c02bd2820

Comment (M-Waszkiewicz-Anaconda) [reply recipe/bld.bat:5]: @JeanChristopheMorinPerso  according to https://learn.microsoft.com/en-us/windows/win32/dlls/dynamic-link-library-creation the dll.lib is used in build process, so if building apps by customers is a valid use case ( which I do not know) we should include it also

Comment (JeanChristopheMorinPerso) [reply recipe/bld.bat:5]: >wasmtime-py just uses dll only, see
https://github.com/AnacondaRecipes/wasmtime-py-feedstock/pull/1/files#diff-6575ceeaf967f9e78afe3221eb5d5034d1edc55ffd549e28e7a1a62c02bd2820

I would look at the builds logs if it was me.

>@JeanChristopheMorinPerso according to https://learn.microsoft.com/en-us/windows/win32/dlls/dynamic-link-library-creation the dll.lib is used in build process, so if building apps by customers is a valid use case ( which I do not know) we should include it also

Yes, and that's why I said "if someone needs these files, then we could just push a PR to rebuild the package"

**Pull request merged**